docs/source/run/service.rst

```diff
@@ -0,0 +1,121 @@
+====================
+Running as a service
+====================
+
+Now that you can run the bot, you probably want it to run forever in the
+background of your system. While there are many ways to do so, there are a few
+steps and requirements that remain the same to properly run Sopel as a service.
+
+
+Requirements to run a service
+=============================
+
+Sopel needs the following:
+
+* a non-root user to run as
+* read/write access to its configuration file
+* a writable pid directory
+  (:attr:`~sopel.config.core_section.CoreSection.pid_dir`)
+* a writable logs directory
+  (:attr:`~sopel.config.core_section.CoreSection.logdir`)
+* a writable data directory
+  (:attr:`~sopel.config.core_section.CoreSection.homedir`)
+
+.. important::
+
+    Do **not** run Sopel as the **root** user. And avoid if possible to run
+    with a privileged user.
+
+
+Running with systemd
+====================
+
+`systemd`__ is a service manager used by Ubuntu and most Debian-derived
+Linux distributions. Here is an example of a service unit file to run Sopel as
+a service, placed at ``/etc/systemd/system/sopel.service``:
+
+.. code-block:: systemd
+
+    [Unit]
+    Description=Sopel IRC bot
+    Documentation=https://sopel.chat/
+    After=network-online.target
+
+    [Service]
+    Type=simple
+    User=sopel
+    RuntimeDirectory=sopel
+    StateDirectory=sopel
+    LogsDirectory=sopel
+    PIDFile=/run/sopel/sopel.pid
+    ExecStart=/home/sopel/.local/bin/sopel start
+    Restart=on-failure
+    RestartPreventExitStatus=2
+    RestartSec=30
+    Environment=LC_ALL=en_US.UTF-8 SOPEL_CONFIG_DIR=/etc/sopel
+
+    [Install]
+    WantedBy=multi-user.target
+
+.. __: https://systemd.io/
+
+For this service to work you will need the following:
+
+* to create a ``sopel`` user
+* to install ``sopel`` with that user (or to change the path in the
+  ``ExecStart``)
+* to place your Sopel configuration file into ``/etc/sopel/default.cfg``, and
+  to setup its access rights so that the ``sopel`` user can read and write it
+* and to edit the ``[core]`` section of this configuration file to set the
+  appropriate paths:
+
+  .. code-block:: ini
+
+      [core]
+      homedir = /var/lib/sopel
+      pid_dir = /run/sopel
+      logdir = /var/log/sopel
```

Comment:
We should consider adding something like `-o core.homedir=...` to `sopel start`, so the service file can override some settings there instead of requiring a specific Sopel config to work correctly.

Reply:
[Environment variables](https://sopel.chat/docs/cli#overriding-individual-settings) already allow this without needing to touch the CLI code.

Reply:
Ah. I don't like the `.`/`_` thing, but I'm inclined to set those vars and not ask config changes then.